lib/irb/history.rb

```diff
@@ -51,7 +51,7 @@ def save_history
 
         if File.exist?(history_file) &&
            File.mtime(history_file) != @loaded_history_mtime
-          history = history[@loaded_history_lines..-1] if @loaded_history_lines
+          @loaded_history_lines&.times { history.delete_at(0) }
```

Comment:
Maybe there's a better way to do this but `Readline::HISTORY` is pretty limited in how it can be manipulated.

The original issue here was a `no implicit conversion of Range into Integer` error because `Readline::HISTORY` doesn't take a range in `[]`.

Reply:
Unlike the previous code, `Reline::HISTORY` will be changed in this operation. Maybe it works, but it's safer and easy to maintain if it doesn't change HISTORY object itself.

Reply:
Good point.  Updated.